File: StoringDataDemo/app/src/main/java/com/example/storingdatademo/MainActivity.java
package com.example.storingdatademo;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;

import java.io.IOException;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "Storage";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        SharedPreferences sharedPreferences = this.getSharedPreferences("com.example.storingdatademo", Context.MODE_PRIVATE);
//        sharedPreferences.edit().putString("username","H3RMIT").apply();
//        String username = sharedPreferences.getString("username","");
        ArrayList<String> friends = new ArrayList<>();
        friends.add("Vishal");
        friends.add("Vijay");
        friends.add("Babul");
        try {
            sharedPreferences.edit().putString("friends", ObjectSerializer.serialize(friends)).apply();
            Log.i(TAG, ObjectSerializer.serialize(friends));
        } catch (IOException e) {
            e.printStackTrace();
        }

        ArrayList<String> newFriends = new ArrayList<>();
        try {
            newFriends = (ArrayList<String>) ObjectSerializer.deserialize(sharedPreferences.getString("friends", ObjectSerializer.serialize(new ArrayList<String>())));
        } catch (Exception e) {
            e.printStackTrace();
        }
        Log.i(TAG, newFriends.toString());
    }
}

File: DynamicFragments/app/src/main/java/com/example/dynamicfragments/MainActivity.kt
package com.example.dynamicfragments

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val bundle = Bundle()
        bundle.putString("Key","H3rmit")
        val fragment = FirstFragment()
        fragment.arguments = bundle

        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager)
        viewPagerAdapter.apply {
            add(fragment)
            add(BlankFragment())
            add(FirstFragment())
        }
        viewPager.adapter = viewPagerAdapter
//        supportFragmentManager
//                .beginTransaction()
//                .replace(R.id.container,fragment)
//                .commitNow()
//        supportFragmentManager
//            .beginTransaction()
//            .replace(R.id.container,BlankFragment())
//            .commitNow()
    }
}